modules/agent-install-networking.adoc

```diff
@@ -0,0 +1,104 @@
+// Module included in the following assemblies:
+//
+// * installing/installing-with-agent/installing-with-agent.adoc
+
+:_content-type: CONCEPT
+[id="agent-install-networking_{context}"]
+= Networking
+
+During the initial boot, the machines require an IP address configuration that is set either through a Dynamic Host Configuration Protocol (DHCP) server or statically by providing the required boot options. After a network connection is established, the machines download their Ignition config files from an HTTP or HTTPS server.
```

Comment:
This IP address is required for a node to identify as node zero. A node zero node runs the necessary services required for the agent-based installer.

**Preferred way (with agent-config.yaml)**
The IP for node zero can be selected via 3 possible use cases.
1. agent-config.yaml just contains `rendezvousIP` field and no `networkConfig` information under `hosts`. In this case, the value of `rendezvousIP` field is used.
2. agent-config.yaml contains `rendezvousIP` field along with `networkConfig` information under `hosts`. In this case, the value of `rendezvousIP` field is used.
3. agent-config.yaml does not contain `rendezvousIP` field but has the `networkConfig` information under `hosts`. In this case, the IP from the first `interfaces` from the list of `interfaces` under `networkConfig` is used.

When neither `rendezvousIP` nor `networkConfig` is provided in agent-config, then the command to create the image fails with error
 ```
failed to fetch Agent Installer ISO: 
failed to fetch dependency of "Agent Installer ISO": 
failed to generate asset "Agent Installer Ignition": 
missing rendezvousIP in agent-config or at least one NMStateConfig manifest
```

**ZTP**

For the ZTP flow, when nmstateconfig is provided then the first IP from the user-provided NMStateConfigs is set as the node0 IP (a.k.a.) rendezvous IP